cdap-master/src/main/java/co/cask/cdap/data/runtime/main/MasterServiceMain.java

```diff
@@ -330,6 +329,9 @@ private LeaderElection createLeaderElection() {
       public void leader() {
         LOG.info("Became leader for master services");
 
+        twillRunner = baseInjector.getInstance(TwillRunnerService.class);
+        twillRunner.startAndWait();
```

Comment:
This doesn't work. `TwillRunner` is a guava service, so it can only started once. You need to create a new instance every time when it becomes leader, meaning you cannot create the twillRunner with the baseInjector (since it's binded as singleton).